### PR TITLE
LibC: Portability improvements

### PR DESCRIPTION
**Pull request by rburchell**
Just a few portability improvements from trying to compile some external code. Note that I haven't actually validated them on anything at runtime, since there were some other deeper (more scary) problems like missing dlfcn.h.

### CODE REVIEW

File: LibC/arpa/inet.cpp [hunk 1]
```diff
@@ -12,6 +14,7 @@ const char* inet_ntop(int af, const void* src, char* dst, socklen_t len)
         return nullptr;
     }
     auto* bytes = (const unsigned char*)src;
+    ASSERT(strlen((const char*)bytes) == INET_ADDRSTRLEN);
```
Comment (awesomekling) [inline LibC/arpa/inet.cpp]: This doesn't seem right, the longest IPv4 address string would be e.g "255.255.255.255" which is 15 characters.
Also, they might be shorter like "1.1.1.1" (7 characters).

Comment (awesomekling) [reply LibC/arpa/inet.cpp]: Oh wait, what is this code event.. isn't "bytes" a raw 32-bit IP address here?

Comment (rburchell) [reply LibC/arpa/inet.cpp]: Sorry, that was completely braindead from reading the manpage and not really reading the manpage. Removed it :)

**Comment by awesomekling** [comment]
Excellent changes! LibC compability grows by a small amount :)

**Pull request merged**